app/controllers/documents_controller.rb

```diff
@@ -25,12 +28,29 @@ def create
   end
   # rubocop:enable Metrics/AbcSize, Metrics/MethodLength
 
-  def download; end
+  def download
+    presign_download = Datastore::Documents::Download.new(document: @document, log_context: log_context).call
+
+    unless presign_download
+      raise Errors::DocumentUnavailable,
+            "Could not download `#{@document.filename}` with object_key: `#{@document.s3_object_key}`"
+    end
+
+    redirect_to(presign_download.url, allow_other_host: true)
+  end
 
   def destroy; end
 
   private
 
+  def set_document
+    @document = Document.find(params[:document_id])
```

Comment:
Can we do the document find after the current_crime_application check (to prevent leaking information in request times).